### ISSUE TITLE
Next Studio folder Failing `next build` due to Webpack errors

### ISSUE THREAD

**Issue by Vinlock** [sanity-io/next-sanity#266]
Hello 👋🏼 ,

I'm using Next 13 with the `appDir`. I have tried this with the `pages` directory as well. When I try to run `next build` the build always fails with:
```
info  - Creating an optimized production build...
--
21:48:20.761 | Failed to compile.
21:48:20.761 |  
21:48:20.762 | ./node_modules/sanity/node_modules/@sanity/ui/dist/index.esm.js
21:48:20.763 | Module parse failed: 'import' and 'export' may appear only with 'sourceType: module' (61:0)
21:48:20.763 | You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
21:48:20.763 | \| function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
21:48:20.767 | \| function _toPrimitive(input, hint) { if (typeof input !== "object" \|\| input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint \|\| "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
21:48:20.767 | > import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
21:48:20.768 | \| import { useMemo, useState, useRef, useEffect, createContext, useContext, useLayoutEffect, forwardRef, useId, useCallback, cloneElement, isValidElement, createElement, Component, memo, useReducer, Children, Fragment as Fragment$1 } from 'react';
21:48:20.768 | \| import ReactIs, { isElement as isElement$1, isFragment, isValidElementType } from 'react-is';
21:48:20.768 |  
21:48:20.769 | Import trace for requested module:
21:48:20.769 | ./node_modules/sanity/node_modules/@sanity/ui/dist/index.esm.js
21:48:20.770 | ./node_modules/sanity/lib/desk.esm.js
21:48:20.770 | ./sanity.config.ts
21:48:20.770 | ./app/studio/[[...index]]/loading.tsx
21:48:20.771 |  
21:48:20.775 |  
21:48:20.776 | > Build failed because of webpack errors
```

Happy to give access to my repo to anyone specific. It is a fairly small repo too.

Though here are the specs atm:

## `app/studio/[[...index]]/head.tsx`
```
import { NextStudioHead } from 'next-sanity/studio/head'

const CustomStudioHead = () => {
	return (
		<>
			<NextStudioHead favicons={false} />
			<link
				rel="icon"
				type="image/png"
				sizes="32x32"
				href="https://www.sanity.io/static/images/favicons/favicon-32x32.png"
			/>
		</>
	)
}

export default CustomStudioHead
```

## `app/studio/[[...index]]/loading.tsx`
```
'use client'

import NextStudioLoading from 'next-sanity/studio/loading'
import config from '../../../sanity.config'

export default () => {
	return <NextStudioLoading config={config} />
}
```

## `app/studio/[[...index]]/page.tsx`
```
'use client'

import { NextStudio } from 'next-sanity/studio'
import config from '../../../sanity.config'

const Page = () => {
	return <NextStudio config={config} />
}

export default Page
```

## `sanity.config.ts`
```
import { visionTool } from '@sanity/vision'
import { defineConfig } from 'sanity'
import { deskTool } from 'sanity/desk'
import { publicConfig } from './config'
import { schemaTypes } from './sanity/schemas'

export default defineConfig({
	basePath: '/studio',

	projectId: publicConfig.sanity.projectId,
	dataset: publicConfig.sanity.dataset,

	plugins: [
		deskTool(),
		visionTool(),
	],

	schema: {
		types: schemaTypes,
	},
})
```

## `publicConfig` is just:
```
export const publicConfig = {
	sanity: {
		projectId: process.env.NEXT_PUBLIC_SANITY_PROJECT_ID as string,
		dataset: process.env.NEXT_PUBLIC_SANITY_DATASET as string,
	},
}
```

## `next.config.js`
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    remotePatterns: [
      {
        hostname: 'cdn.sanity.io',
      },
    ],
  },
  redirects: async () => {
    return [
      {
        source: '/aion/schedule',
        destination: '/',
        permanent: false,
      },
      {
        source: '/aion/schedule/america',
        destination: '/aion/schedule/aion-america-schedule',
        permanent: false,
      },
    ]
  },
  experimental: {
    appDir: true,
  },
}

module.exports = nextConfig
```

Anything I'm missing?

**Comment by mariuslundgard** [comment]
Hey, which version of `@sanity/ui` are you using?

**Comment by Vinlock** [comment]
> Hey, which version of `@sanity/ui` are you using?

Here are the versions of all the packages I am using.

```
"@sanity/image-url": "1.0.1",
"@sanity/ui": "1.0.12",
"@sanity/vision": "3.1.4",
"next-sanity": "3.1.9",
"sanity": "3.1.4"
```

**Comment by Vinlock** [comment]
> Hm, odd, it works fine here with the same versions. Have you tried wiping `node_modules`, lock files, and re-installing?
> 
> ```shell
> rm -rf node_modules yarn.lock package-lock.json pnpm-lock.yaml
> ```
> 
> Full disclosure: we saw the error you describe when using `@sanity/ui@1.0.6` but it was fixed in `@sanity/ui@1.0.7`.

That works 🤦🏼 lmao. Thank you! I must've had `1.0.6` installed and it didn't update properly due to `yarn.lock` or something...
